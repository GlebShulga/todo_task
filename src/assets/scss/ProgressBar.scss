.container {
    height: 20%;
    width: 90%;
    background-color: #e0e0de;
    border-radius: 50px;
    margin: 0.5rem;
}

.filler {
    height: 100%;
    background-color: black;
    border-radius: inherit;
    text-align: right;
}

.label {
    padding: 0.5rem;
    color:white;
    font-weight: bold;
}