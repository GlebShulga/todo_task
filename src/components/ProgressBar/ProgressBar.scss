.progress-bar {
    width: 90%;
    border-radius: 5rem;
    margin: 0.5rem 1rem;
}
