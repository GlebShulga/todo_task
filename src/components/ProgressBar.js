import React from "react";
import "../assets/scss/ProgressBar.scss";

const ProgressBar = ({ completed }) => {

  return (
    <div className="container">
      <div style={{ width: `${completed}%` }} className="filler">
        <span className="label">{`${completed}%`}</span>
      </div>
    </div>
  );
};

export default ProgressBar;
