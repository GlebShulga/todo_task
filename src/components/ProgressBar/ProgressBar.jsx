import React from "react";
import { useSelector } from "react-redux";
import "./ProgressBar.scss";

const ProgressBar = () => {
  const { taskList } = useSelector((s) => s.task);
  const { chosenCategory } = useSelector((s) => s.category);

  let isDone = 0;
  let count = 0;
  taskList
    ?.filter((item) => item.categoryId === chosenCategory?.categoryId)
    .forEach((item) => {
      if (item.status === "done") {
        isDone += 1;
      }
      count += 1;
    });

  const result = count > 0 ? Math.round((isDone / count) * 100) : 100;

  return (
    <progress value={result} max="100" className="progress-bar"></progress>
  );
};

export default ProgressBar;
