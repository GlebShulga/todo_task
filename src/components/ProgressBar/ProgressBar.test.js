import React from "react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import { shallow } from "enzyme";
import ProgressBar from "./ProgressBar";

let mockStore;
const mockStoreConf = configureStore([]);

describe("ProgressBar Component", () => {
  beforeEach(() => {
    mockStore = mockStoreConf({
      category: { isFilterStatusDone: false },
      task: {
        taskList: [
          {
            categoryId: "3",
            taskId: "1",
            title: "Just do it",
            description: "bla",
            status: "new",
            _createdAt: 1630474050422,
          },
          {
            categoryId: "3",
            taskId: "2",
            title: "Do it one more time",
            status: "new",
            description: "",
            _createdAt: 1630474050402,
          },
        ],
      },
    });
  });
  it("should render ProgressBar Component", () => {
    shallow(
      <Provider store={mockStore}>
        <ProgressBar />
      </Provider>
    );
  });

  it("ProgressBar snapshot", () => {
    const component = renderer.create(
      <Router>
        <Provider store={mockStore}>
          <ProgressBar />
        </Provider>
      </Router>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
