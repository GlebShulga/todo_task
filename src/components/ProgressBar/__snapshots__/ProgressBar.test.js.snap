// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar Component ProgressBar snapshot 1`] = `
<progress
  className="progress-bar"
  data-testid="ProgressBar"
  max="100"
  value={100}
/>
`;
